src/models/spaces/breeding/breeding.ts

```diff
@@ -30,14 +30,39 @@ const NestPair = types.model({
 export type INestPair = Instance<typeof NestPair>;
 
 export function createBreedingModel(breedingProps: any) {
-  breedingProps.nestPairs = [
-    {leftMouse: {sex: "male", genotype: "RC"}, rightMouse: {sex: "female", genotype: "RR"}, label: "Pair 1"},
-    {leftMouse: {sex: "male", genotype: "CC"}, rightMouse: {sex: "female", genotype: "RR"}, label: "Pair 2"},
-    {leftMouse: {sex: "male", genotype: "CC"}, rightMouse: {sex: "female", genotype: "CC"}, label: "Pair 3"},
-    {leftMouse: {sex: "male", genotype: "RR"}, rightMouse: {sex: "female", genotype: "RR"}, label: "Pair 4"},
-    {leftMouse: {sex: "male", genotype: "RC"}, rightMouse: {sex: "female", genotype: "RC"}, label: "Pair 5"},
-    {leftMouse: {sex: "male", genotype: "CC"}, rightMouse: {sex: "female", genotype: "RC"}, label: "Pair 6"},
-  ];
+  if (!breedingProps.nestPairs || breedingProps.nestPairs.length === 0) {
+    const numPairs = 6;
+    const sexpairs = [];
+    for (let i = 0; i < numPairs; i++) {
+      Math.random() >= 0.5 ? sexpairs.push({ls: "male", rs: "female"}) : sexpairs.push({ls: "female", rs: "male"});
+    }
+    breedingProps.nestPairs = [
+      { leftMouse: {sex: sexpairs[0].ls, genotype: "RC"},
+        rightMouse: {sex: sexpairs[0].rs, genotype: "RR"},
+        label: "Pair 1"
+      },
+      { leftMouse: {sex: sexpairs[1].ls, genotype: "CC"},
+        rightMouse: {sex: sexpairs[1].rs, genotype: "RR"},
+        label: "Pair 2"
+      },
+      { leftMouse: {sex: sexpairs[2].ls, genotype: "CC"},
+        rightMouse: {sex: sexpairs[2].rs, genotype: "CC"},
+        label: "Pair 3"
+      },
+      { leftMouse: {sex: sexpairs[3].ls, genotype: "RR"},
+        rightMouse: {sex: sexpairs[3].rs, genotype: "RR"},
+        label: "Pair 4"
+      },
+      { leftMouse: {sex: sexpairs[4].ls, genotype: "RC"},
+        rightMouse: {sex: sexpairs[4].rs, genotype: "RC"},
+        label: "Pair 5"
+      },
+      { leftMouse: {sex: sexpairs[5].ls, genotype: "CC"},
+        rightMouse: {sex: sexpairs[5].rs, genotype: "RC"},
+        label: "Pair 6"
+      },
+    ];
+  }
```

Comment:
This is fine, and could go in as-is. I wonder, however, if this is clearer:

```
const breedingPairs = [
      ["RC", "RR"],
      ["CC", "RR"],
      ["CC", "CC"],
      ["RR", "RR"],
      ["RC", "RC"],
      ["CC", "RC"]
    ];

    breedingProps.nestPairs = breedingPairs.map((pair, i) => {
      const leftSex = Math.round(Math.random());
      const rightSex = 1 - leftSex;
      return {
        leftMouse: {sex: leftSex, genotype: pair[0]},
        rightMouse: {sex: rightSex, genotype: pair[1]},
        label: `Pair ${i + 1}`
      };
    });
```

To me this is clearer, because the substantive part is purely `breedingPairs`, and this clarifies what they are. This array could also be easily brought in via authoring if we ever wanted to author the colors, or randomized if we wanted to randomize the placement etc.

However, the final result is the same, so whichever is clearer to you.

(Note above is aircoded, so there may be syntax or logic errors.)

Reply:
Yeah, this is a nice approach.  I'll rough this in and push another change.  Thanks!

Reply:
Changes pushed.